lib/bom.gradle

```diff
@@ -12,16 +12,18 @@ configure(projectsWithFlags('bom')) {
         }
     }
 
-    dependencies {
-        constraints {
-            projectsWithFlags('java', 'publish').toList().sort { a, b ->
-                def groupComparison = "${a.group}".compareTo("${b.group}")
-                if (groupComparison != 0) {
-                    return groupComparison
+    afterProjectsWithFlag('java') { Set<Project> projects ->
```

Comment:
I don't know why but the callbacks of `afterProjectsWithFlag()` was not called in the tasks' lifecycle.
It worked correctly when using `afterEvaluate()` and `projectsWithFlags()` separately.
https://github.com/line/armeria/pull/4377/files#diff-99c9f9c14309adeee494c305b31bb4774d339f731c426330858d51e3190a0f30R131-R132

Reply:
Are you suggesting to change like this?

```diff
diff --git a/lib/bom.gradle b/lib/bom.gradle
index 97e34ca..bb6154c 100644
--- a/lib/bom.gradle
+++ b/lib/bom.gradle
@@ -12,9 +12,9 @@ configure(projectsWithFlags('bom')) {
         }
     }
 
-    afterProjectsWithFlag('java') { Set<Project> projects ->
-        dependencies {
-            constraints {
+    dependencies {
+        constraints {
+            afterEvaluate {
                 projectsWithFlags('java', 'publish').toList().sort { a, b ->
                     def groupComparison = "${a.group}".compareTo("${b.group}")
                     if (groupComparison != 0) {

```

Reply:
Correct. It worked fine for my cases.

Reply:
I see, sorry for late but applied your comment at https://github.com/line/gradle-scripts/pull/124/commits/1bc04d40e42d7620097507cf2af7c3dfd2ea9e10

Reply:
Did it work as you expected? I wanted to know if this PR is ready to merge.

Reply:
@ikhoon Yes, it worked well. Sorry for late reply

Reply:
I'm glad it's going well. 😀